proof/taproot.go

```diff
@@ -292,7 +360,16 @@ func tapLeafHash(preimage []byte) (*chainhash.Hash, error) {
 	}
 
 	version := txscript.TapscriptLeafVersion(preimage[0])
-	script := preimage[1:]
+
+	// The script is encoded with a leading VarByte that indicates its total
```

Comment:
Something else we need to add to the spec? 

But on the TLV level don't we already know what this length is, which is why we're able to parse it in the first place?